osu.Game/Overlays/Dashboard/Friends/FriendsLayout.cs

```diff
@@ -0,0 +1,252 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Users;
+using osuTK;
+
+namespace osu.Game.Overlays.Dashboard.Friends
+{
+    public class FriendsLayout : CompositeDrawable
+    {
+        private List<User> users = new List<User>();
+
+        public List<User> Users
+        {
+            get => users;
+            set
+            {
+                request?.Cancel();
```

Comment:
Why does the setter cancel the request? As far as I can see it's only fired once in `LoadComplete`, and the setter is only used when the request *succeeds*? (Leaving aside the fact that a setter doing API logic is in itself a bit of a code smell.)

Reply:
It's just to make tests work.
Probably exposing a method like `Fetch` will be a better approach (?)